web/src/components/molecules/Common/ResizableProTable/index.tsx

```diff
@@ -0,0 +1,87 @@
+import styled from "@emotion/styled";
+import { useEffect, useState } from "react";
+
+import ProTable, {
+  ProColumns,
+  OptionConfig,
+  ListToolBarProps,
+} from "@reearth-cms/components/atoms/ProTable";
+import { ResizableTitle } from "@reearth-cms/components/molecules/Common/ResizableProTable/resizble";
+import type { ResizeCallbackData } from "@reearth-cms/components/molecules/Common/ResizableProTable/resizble";
+
+export type Props = {
+  className?: string;
+  dataSource?: any[];
+  proColumns?: ProColumns<any, "text">[];
+  loading: boolean;
+  options?: false | OptionConfig | undefined;
+  toolbar?: ListToolBarProps | undefined;
+};
+
+const ResizableProTable: React.FC<Props> = ({
+  dataSource,
+  proColumns,
+  loading,
+  options,
+  toolbar,
+}) => {
+  const [columns, setColumns] = useState<ProColumns<any, "text">[]>([]);
+
+  useEffect(() => {
+    if (Array.isArray(proColumns)) {
+      setColumns(proColumns);
+    }
+  }, [proColumns, setColumns]);
+
+  const handleResize =
+    (index: number) =>
+    (_: React.SyntheticEvent<Element>, { size }: ResizeCallbackData) => {
+      const newColumns = [...columns];
+      newColumns[index] = {
+        ...newColumns[index],
+        width: size.width,
+      };
+      setColumns(newColumns);
+    };
+
+  const mergeColumns: ProColumns<any, "text">[] = columns?.map((col, index) => ({
+    ...col,
+    onHeaderCell: column => ({
+      minWidth: (column as ProColumns<any, "text"> & { minWidth: number }).minWidth,
+      width: (column as ProColumns<any, "text">).width,
+      onResize: handleResize(index),
+    }),
+  }));
+
+  return proColumns ? (
+    <Wrapper>
+      <ProTable
+        dataSource={dataSource}
+        columns={mergeColumns}
+        components={{
+          header: {
+            cell: ResizableTitle,
+          },
+        }}
+        rowKey="id"
+        search={false}
+        loading={loading}
+        toolbar={toolbar}
+        options={options}
+        tableStyle={{ overflowX: "scroll" }}
+      />
+    </Wrapper>
+  ) : null;
```

Comment:
Do you want to return null? I thought we would want to still show the table even if no values and let the table display the antd `No data` message like the other tables, no?

Reply:
@KaWaite , this condition checks if there is a `columns` definition (not data), so if the user did not specify the columns of the table, I should not show the table at all.
In case we are fetching columns definition from the backend, this condition is to wait for columns before rendering.

Reply:
Wouldn't we want to do that kind of checking in the parent component? That way, we can keep this component simpler and show Loading if its loading as well.

Reply:
@KaWaite, totally right.
Changed and I kept the condition in the parent component.